modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -429,9 +429,12 @@ void ONNXImporter::populateNet(Net dstNet)
         }
         else if (layer_type == "Add" || layer_type == "Sum")
         {
-            if (layer_id.find(node_proto.input(1)) == layer_id.end())
+            bool is_const_0 = layer_id.find(node_proto.input(0)) == layer_id.end();
+            bool is_const_1 = layer_id.find(node_proto.input(1)) == layer_id.end();
+            if (is_const_0 ^ is_const_1)
             {
-                Mat blob = getBlob(node_proto, constBlobs, 1);
+                int blob_id = is_const_0 ? 0 : 1;
+                Mat blob = getBlob(node_proto, constBlobs, blob_id);
```

Comment:
`Mat blob = getBlob(node_proto, constBlobs, is_const_0 ? 0 : 1);` also looks fine as for me.